eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/LongInterval.java

```diff
@@ -530,9 +530,9 @@ public ImmutableLongList distinct()
     }
 
     @Override
-    public ImmutableLongList subList(int fromIndex, int toIndex)
+    public LongInterval subList(int fromIndex, int toIndex)
     {
-        throw new UnsupportedOperationException("subList not yet implemented!");
+        return new LongInterval(fromIndex, toIndex, this.step);
```

Comment:
this looks wrong. The fromIndex/toIndex of the list is not the same as the from/to of the interval.